fc-community-contracts/src/FactChainNFT.sol

```diff
@@ -5,29 +5,45 @@ import {ERC721URIStorage} from "openzeppelin-contracts/contracts/token/ERC721/ex
 import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
 import {Ownable} from "./utils/Ownable.sol";
 
+contract FactChainSFT {
+    function mint(address[] memory raters, string memory ipfsHash, uint256 id) public returns (uint256) {}
+}
+
 /// @title FactChainers Community Notes NFT
 /// @author Pierre HAY
 /// @notice
 /// @dev
 contract FactChainNFT is Ownable, ERC721URIStorage {
+    FactChainSFT factsLithography;
     uint256 public tokenIdCounter;
     string private baseTokenURI;
 
-    constructor(address _owner, string memory _baseTokenURI) Ownable(_owner) ERC721("FactchainNotes", "FCN") {
+    constructor(address _owner, address _factsLitography, string memory _baseTokenURI)
+        Ownable(_owner)
+        ERC721("FactchainNotes", "FCN")
+    {
         tokenIdCounter = 0;
         baseTokenURI = _baseTokenURI;
+        factsLithography = FactChainSFT(_factsLitography);
     }
 
     /// @notice mint a NFT
-    /// @param recipient destination address
-    function mint(address recipient, string memory ipfsHash) public onlyOwner returns (uint256) {
-        // An overflow here would be a great collective achievement (2**256 factchain notes)
+    /// @param creator destination address for the orginal fact (NFT-721)
+    /// @param raters raters's addresses to receive a copy of the original (SFT-1155)
+    /// @param ipfsHash metadata ID on ipfs shared by the original and its copies
+    function mint(address creator, address[] memory raters, string memory ipfsHash)
```

Comment:
I think that this function should accept `postUrl` as well and build an nftIds map of `(postUrl + creator) -> id`